pkg/chart/chart.go

```diff
@@ -268,8 +268,9 @@ func ParseDetails(data []byte) (*Details, error) {
 // Depending on the repo namespace and the
 func GetAppRepoAndRelatedSecrets(appRepoName, appRepoNamespace string, handler kube.AuthHandler, userAuthToken, cluster, kubeappsNamespace string) (*appRepov1.AppRepository, *corev1.Secret, *corev1.Secret, error) {
 	client, err := handler.AsUser(userAuthToken, cluster)
-	if kubeappsNamespace == appRepoNamespace {
-		// If we're parsing a global repository (from the kubeappsNamespace), use a service client.
+	if cluster == "" || kubeappsNamespace == appRepoNamespace {
+		// If we're parsing a global repository (from the kubeappsNamespace),
+		// or we don't specify any cluster, use a service client.
```

Comment:
I think the key point here is that we use a service account if we're fetching a global AppRepository, which is currently one in the Kubeapps namespace. Given that we currently only support app repositories on the cluster on which Kubeapps is installed, is going to be either:
1. The app repository explicitly matches because `cluster` is the kubeapps cluster (unfortunately we don't currently have access to the kubeapps cluster here[1]) and the app repo namespace is the kubeapps namespace, or 
2. The cluster and kubeappsNamespace are both empty because the UI `clusters` config doesn't include the cluster on which kubeapps is installed in which case we must always be fetching global app repos (nothing else is supported).

With this in mind, I'd use variable names to document this a little more clearly with (and to be clear, this is fixing the bug introduced, not a good longer-term solution):

```suggestion
	// If the UI clusters configuration did not include the cluster on which Kubeapps is installed then
	// we won't know the kubeappsNamespace but it will be empty. In this scenario Kubeapps only supports
	// global app repositories (#1982 ).
	nonUIKubeappsCluster = (cluster == "" && kubeappsNamespace == "")
	if nonUIKubeappsCluster || kubeappsNamespace == appRepoNamespace {
		// If we're parsing a global repository then use a service client.
```

[1] We should fix this as there are security implications, in that if I create a namespace on a different cluster with the same name as that in which Kubeapps is installed, we'd currently use the service account. We could possibly just pass `kubeappsCluster` to this function, if it's available at the call-site, not sure.